services/s3/src/it/java/software/amazon/awssdk/services/s3/S3PresignerIntegrationTest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.time.Duration;
+import java.util.UUID;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.HttpExecuteRequest;
+import software.amazon.awssdk.http.HttpExecuteResponse;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.services.s3.model.RequestPayer;
+import software.amazon.awssdk.services.s3.presigner.S3Presigner;
+import software.amazon.awssdk.services.s3.presigner.model.PresignedGetObjectRequest;
+import software.amazon.awssdk.services.s3.utils.S3TestUtils;
+import software.amazon.awssdk.utils.IoUtils;
+
+public class S3PresignerIntegrationTest {
+    private static S3Client client;
+    private static S3Presigner presigner;
+    private static String testBucket;
+    private static String testNonDnsCompatibleBucket;
+    private static String testObjectKey;
+    private static String testObjectContent;
+
+    @BeforeClass
+    public static void setUp() {
+        client = S3Client.create();
+        presigner = S3Presigner.create();
+        testBucket = S3TestUtils.getTestBucket(client);
+        testNonDnsCompatibleBucket = S3TestUtils.getNonDnsCompatibleTestBucket(client);
+        testObjectKey = "s3-presigner-it-" + UUID.randomUUID();
+        testObjectContent = "Howdy!";
```

Comment:
These should still be per-test (@Before)

Reply:
+1